### PR TITLE
Fix #3331 - ACL

### PR DESCRIPTION
**Pull request by rana-nouman**
http://dev.projects4.me/app/project/cb4567905acd-06d6-1141-7dd99b59dd21/issue/3331

### CODE REVIEW

File: core/mvc/controllers/RestController.php [hunk 1]
```diff
@@ -14,6 +14,7 @@
 use Gaia\Libraries\Utils\Util;
 use Phalcon\Events\ManagerInterface as EventsManagerInterface;
 use function Gaia\Libraries\Utils\create_guid;
+use phpDocumentor\Parser\Exception;
```
Comment (gollomer) [inline core/mvc/controllers/RestController.php]: Why is this required?

File: core/mvc/controllers/RestController.php [hunk 2]
```diff
@@ -126,13 +127,34 @@ class RestController extends \Phalcon\Mvc\Controller implements \Phalcon\Events\
      * @var array $aclMap
      */
     protected $aclMap = array(
```
Comment (gollomer) [inline core/mvc/controllers/RestController.php:127]: Wouldn't it be better to move this to the Permissions model?

File: autoload.php [hunk 3]
```diff
@@ -70,6 +69,7 @@
         "Gaia\\Core\\MVC\\REST\\Controllers" => APP_PATH. '/core/mvc/controllers/',
         "Gaia\\Core\\MVC\\Models\\Relationships\\Factory" => APP_PATH. '/core/mvc/models/relationships/factory',
         "Gaia\\Core\\MVC\\Models\\Query" => APP_PATH. '/core/mvc/models/query/',
+        "Tests\\Gaia\\Acl" => APP_PATH. '/tests/acl/'
```
Comment (gollomer) [inline autoload.php]: I think the Autoload for tests should be seperate.

**Review by gollomer — CHANGES_REQUESTED** [review]

File: core/mvc/controllers/RestController.php (hunk 1 above)
Comment (rana-nouman) [reply core/mvc/controllers/RestController.php]: Removed! It was a typo mistake by VSCODE.

File: core/mvc/controllers/RestController.php (hunk 2 above)
Comment (rana-nouman) [reply core/mvc/controllers/RestController.php:127]: This is the REST controller's action related mapping and that should be done in RestController instead of doing It in Permissions model.

File: autoload.php (hunk 3 above)
Comment (rana-nouman) [reply autoload.php]: Created a separate autoloader for testing purpose

**Review by gollomer — APPROVED** [review]
